src/core/tokenizer.c

```diff
@@ -195,3 +197,78 @@ tokenize(filter_pipeline_t *fp, const char *text, size_t text_len __unused)
 	free(content);
 	return NULL;
 }
+
+
+
+/*
+ * tokenize: split the text into tokens separated by a whitespace
+ * and pass the token through the filter pipeline.
+ */
+tokenset_t *
+tokenize2(filter_pipeline_t *fp, const char *text, size_t text_len __unused)
+{
+
+	UBreakIterator *it_token;
+	UChar          utext[text_len + 1]; 
```

Comment:
You must allocate this buffer dynamically. Although the default stack size on modern UNIX-like systems is pretty large (e.g. on Linux it's in excess of megabytes), the user can overflow it by just passing a very large string.